src/auth/ui/vue/show.ts

```diff
@@ -40,25 +33,33 @@ import {
     ExtensionUse,
     showConnectionsPageCommand,
     addConnection,
+    hasIamCredentials,
+    hasBuilderId,
+    hasSso,
+    BuilderIdKind,
 } from '../../utils'
 import { Region } from '../../../shared/regions/endpoints'
 import { CancellationError } from '../../../shared/utilities/timeoutUtils'
 import { validateSsoUrl, validateSsoUrlFormat } from '../../sso/validation'
-import { debounce } from '../../../shared/utilities/functionUtils'
-import { AuthError, ServiceItemId, isServiceItemId, userCancelled } from './types'
+import { AuthError, ServiceItemId, isServiceItemId, authFormTelemetryMapping, userCancelled } from './types'
 import { awsIdSignIn } from '../../../codewhisperer/util/showSsoPrompt'
 import { connectToEnterpriseSso } from '../../../codewhisperer/util/getStartUrl'
 import { trustedDomainCancellation } from '../../sso/model'
 import { FeatureId, CredentialSourceId, Result, telemetry } from '../../../shared/telemetry/telemetry'
-import { AuthFormId, isBuilderIdAuth } from './authForms/types'
+import { AuthFormId } from './authForms/types'
 import { handleWebviewError } from '../../../webviews/server'
-import { cwQuickPickSource, cwTreeNodeSource } from '../../../codewhisperer/commands/types'
 import { Commands, VsCodeCommandArg, placeholder, vscodeComponent } from '../../../shared/vscode/commands2'
+import { CodeWhispererSource, cwQuickPickSource, cwTreeNodeSource } from '../../../codewhisperer/commands/types'
+import { ClassToInterfaceType } from '../../../shared/utilities/tsUtils'
+import { throttle } from 'lodash'
+import { submitFeedback } from '../../../feedback/vue/submitFeedback'
 
 export class AuthWebview extends VueWebview {
     public override id: string = 'authWebview'
     public override source: string = 'src/auth/ui/vue/index.js'
-    public readonly onDidConnectionUpdate = new vscode.EventEmitter<undefined>()
+    public readonly onDidConnectionChangeCodeCatalyst = new vscode.EventEmitter<void>()
+    public readonly onDidConnectionChangeExplorer = new vscode.EventEmitter<void>()
+    public readonly onDidConnectionChangeCodeWhisperer = new vscode.EventEmitter<void>()
```

Comment:
This looks less ergonomic than the old single emitter. Why can't the handlers decide based on an event arg like `type='codecatalyst'`? When we spread things out it makes the code harder to follow.

Reply:
I had that way initially, but with the current way webviews are setup it caused the event to be emitted 3 times instead of 1. I'll look in to a better solution once I get time.